install/collecting_ncn_mac_addresses.md

```diff
@@ -70,41 +73,48 @@ For help with either of those, see [LiveCD Setup](bootstrap_livecd_remote_iso.md
     ncn-w003-mgmt
     ```
 
-3. Set the nodes to PXE boot and (re)start them.
+1. Set the nodes to PXE boot and start (or restart) them.
 
     ```bash
-    pit# export USERNAME=root
+    pit# USERNAME=root
```

Comment:
Exporting it just determines if it gets passed to child processes. I don't think you need it for xargs, and I don't see how it will matter in terms of when an admin resumes an install. exporting it doesn't put it into /etc/envrironment or anything, so it won't be present if a different shell is started.

```console
hermod-ncn-m001-pit:~ # echo $USERNAME

hermod-ncn-m001-pit:~ # USERNAME=root
hermod-ncn-m001-pit:~ # export IPMI_PASSWORD=initial0
hermod-ncn-m001-pit:~ # grep -o 'ncn-w00[1-3]-mgmt' /etc/dnsmasq.d/statics.conf | sort -u | xargs -I {} ipmitool -U $USERNAME -E -I lanplus -H {} power status
Chassis Power is on
Chassis Power is on
Chassis Power is on
```

Reply:
```
redbull-ncn-m001-pit:~ # username=root
redbull-ncn-m001-pit:~ # IPMI_PASSWORD=initial0
redbull-ncn-m001-pit:~ # grep -oP "($mtoken|$stoken|$wtoken)" /etc/dnsmasq.d/statics.conf | sort -u | xargs -t -i ipmitool -I lanplus -U $username -E -H {} power status
ipmitool -I lanplus -U root -E -H ncn-m002-mgmt power status
Unable to read password from environment
Password:
redbull-ncn-m001-pit:~ # grep -oP "($mtoken|$stoken|$wtoken)" /etc/dnsmasq.d/statics.conf | sort -u | xargs -I ipmitool -I lanplus -U $username -E -H {} power status
xargs: invalid option -- 'U'
Try 'xargs --help' for more information.
```

When I try it with `-t -i` or `-I` xargs isn't resolving `username` or `IPMI_PASSWORD`.

Reply:
IPMI_PASSWORD does need to be exported. That is because we call ipmitool with the -E flag, which tells it to read the variable from the environment. That variable will only be set in the ipmitool environment if it has been exported. That's why I do not remove the export directive for IPMI_PASSWORD, and that explains why your first ipmitool command failed.

The second command failed because you omitted the "{}" after the "xargs -I". Try adding that and see if it works.

Reply:
```bash
hermod-ncn-m001-pit:~ # echo $DOGFOOD

hermod-ncn-m001-pit:~ # DOGFOOD=root
hermod-ncn-m001-pit:~ # grep -o 'ncn-w00[1-3]-mgmt' /etc/dnsmasq.d/statics.conf | sort -u | xargs -i -t ipmitool -U $DOGFOOD  -E -I lanplus -H {} power status
ipmitool -U root -E -I lanplus -H ncn-w001-mgmt power status
Chassis Power is on
ipmitool -U root -E -I lanplus -H ncn-w002-mgmt power status
Chassis Power is on
ipmitool -U root -E -I lanplus -H ncn-w003-mgmt power status
Chassis Power is on
```

Reply:
Nevermind. I am not sure how I saw `xargs -I` being used. It isn't in this PR.